File: src/main/java/com/camackenzie/exvi/core/ExerciseType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core;

/**
 *
 * @author callum
 */
public enum ExerciseType {
    Strength,
    Warmup,
    Cooldown,
    Plyometric,
    Conditioning,
    Powerlifting;

    public static ExerciseType fromString(String s) {
    }
}

File: exvi-core/src/commonMain/kotlin/com/camackenzie/exvi/core/model/ExerciseType.kt
/*
 * Copyright (c) Callum Mackenzie 2022.
 */

package com.camackenzie.exvi.core.model

import com.camackenzie.exvi.core.util.EnumUtils
import com.camackenzie.exvi.core.util.EnumUtils.formatName

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
@Suppress("unused")
enum class ExerciseType {
    Strength,
    Warmup,
    Cooldown,
    Plyometric,
    Conditioning,
    PowerLifting;

    override fun toString(): String = formatName(super.toString())

    companion object {
        @kotlin.jvm.JvmStatic
        fun fromString(s: String): ExerciseType? = EnumUtils.enumFromString<ExerciseType>(s)
    }
}